pkg/network/ebpf/c/tracer/events.h

```diff
@@ -53,12 +53,30 @@ static __always_inline void cleanup_conn(void *ctx, conn_tuple_t *tup, struct so
     u32 *retrans = NULL;
     bool is_tcp = get_proto(&conn.tup) == CONN_TYPE_TCP;
     bool is_udp = get_proto(&conn.tup) == CONN_TYPE_UDP;
+    bool cst_flushable = false;
+
+    cst = bpf_map_lookup_elem(&conn_stats, &(conn.tup));
+    // if we were able to delete the entry, it signals that no other threads have flushed it
+    if (cst && (bpf_map_delete_elem(&conn_stats, &(conn.tup)) == 0)) {
+        cst_flushable = true;
+        conn.conn_stats = *cst;
+    }
+
+    if (is_udp && !cst_flushable) {
+        increment_telemetry_count(udp_dropped_conns);
+        return;
+    }
 
     if (is_tcp) {
         tst = bpf_map_lookup_elem(&tcp_stats, &(conn.tup));
-        if (tst) {
+        if (tst && (bpf_map_delete_elem(&tcp_stats, &(conn.tup)) == 0)) {
             conn.tcp_stats = *tst;
-            bpf_map_delete_elem(&tcp_stats, &(conn.tup));
+        } else {
+            // flush empty connections in the case of tcp failures
+            if (tcp_failure_reason == 0 && !cst_flushable) {
```

Comment:
Not very sure this will not lead to a connection being reported twice. What if we put an empty `tcp_stats_t` for failures? Would you be able to get rid of `tcp_failure_reason`?

Reply:
Isn't there a `failure_reason` field in `tcp_stats_t`?

Reply:
discussed offline I will be inserting the failure reason into tcp_stats during the relevant probe before this, then removing this block